src/controllers/action-queue/runtime.js

```diff
@@ -6,15 +6,36 @@ import * as React from 'react'
 
 import { AirshipToast } from '../../components/common/AirshipToast'
 import { Airship } from '../../components/services/AirshipInstance'
+import { type PushEvent } from '../../controllers/push/types'
 import { queryBorrowPlugins } from '../../plugins/helpers/borrowPluginHelpers'
 import { getCurrencyCode } from '../../util/CurrencyInfoHelpers'
 import { snooze } from '../../util/utils'
+import { preparePushEvents } from './push'
 import { type ActionEffect, type ActionProgram, type ActionProgramState, type ExecutionResult, type ExecutionResults } from './types'
 
 export const executeActionProgram = async (account: EdgeAccount, program: ActionProgram, state: ActionProgramState): Promise<ExecutionResults> => {
-  const { effect } = state
+  let { effect } = state
+  let pushEvents: PushEvent[] = []
+
+  // Dry Run
+  if (effect != null) {
+    const precomputed: ExecutionResult[] = []
+    while (true) {
+      const result = await executeActionOp(account, program, state, true)
+
+      // Stop if action isn't present, meaning we couldn't dryrun
+      if (result.action == null) break
+
+      precomputed.push(result)
+    }
+    // Convert each precomputed operation from the dryrun into an arrya of
```

Comment:
"arrya" sounds like some super-cool sci-fi thing.

Reply:
Haha, thanks for the spell-check catch.